src/test/java/com/yelp/nrtsearch/server/grpc/MergeBehaviorTests.java

```diff
@@ -0,0 +1,405 @@
+/*
+ * Copyright 2020 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.grpc;
+
+import static com.yelp.nrtsearch.server.grpc.GrpcServer.rmDir;
+import static com.yelp.nrtsearch.server.grpc.LuceneServerTest.RETRIEVED_VALUES;
+import static org.junit.Assert.*;
+
+import com.google.common.collect.Sets;
+import com.yelp.nrtsearch.server.LuceneServerTestConfigurationFactory;
+import com.yelp.nrtsearch.server.config.LuceneServerConfiguration;
+import com.yelp.nrtsearch.server.luceneserver.GlobalState;
+import io.grpc.testing.GrpcCleanupRule;
+import io.prometheus.client.CollectorRegistry;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+/**
+ * These tests verify the behavior of segment merges on the searchers and on the segment files. For
+ * old searchers to be pruned: 1. 60 seconds must pass after the searcher stopped being the live
+ * searcher 2. if opened with a snapshot, the snapshot must be released
+ *
+ * <p>For old segment files (which have been merged into new segments) to be deleted: 1. Any
+ * searchers referencing the segments must be pruned 2. If the segments were committed, a new commit
+ * after the merge must be issued
+ */
+public class MergeBehaviorTests {
```

Comment:
If these tests are "slow" could we do something similar to what we did with `YelpReviewsTest`
https://github.com/Yelp/nrtsearch/blob/master/build.gradle#L127

We typically dont run these by default when one invokes: "./gradlew test" but we do run them on travis and if they fail the build fails on travis:
https://github.com/Yelp/nrtsearch/blob/master/.travis.yml#L14